habitat-lab/habitat/datasets/rearrange/rearrange_generator.py

```diff
@@ -519,8 +578,9 @@ def generate_single_episode(self) -> Optional[RearrangeEpisode]:
         for sampler, (sampler_name, num_targets) in zip(
             self._target_samplers.values(), target_numbers.items()
         ):
-            new_goal_receptacles = []
-            for _ in range(num_targets):
+            new_goal_receptacles = []  # type: ignore
+            # for _ in range(num_targets):
+            while len(new_goal_receptacles) < num_targets:
```

Comment:
Can this get stuck in an infinite loop? Can we have a termination due to trying too many times just in case?